src/main/java/org/broadinstitute/hellbender/tools/copynumber/gcnv/GCNVPostProcessor.java

```diff
@@ -0,0 +1,197 @@
+    private final SAMSequenceDictionary samSequenceDictionary;
+
+    /**
+     * We store the previously analyzed record for validating that records are ordered according to
+     * the sequence dictionary
+     */
+    private CopyNumberPosteriorLocatableRecord lastPosteriorRecord = null;
+
+    /**
+     * gCNV postprocessor constructor
+     *
+     * @param outputWriter variant context writer
+     * @param integerCopyNumberStateCollection collection of copy number states considered by post processor
+     * @param sampleName sample name
+     * @param samSequenceDictionary sequence dictionary used for interval order validation
+     */
+    GCNVPostProcessor(final VariantContextWriter outputWriter,
+                                final IntegerCopyNumberStateCollection integerCopyNumberStateCollection,
+                                final String sampleName,
+                                final SAMSequenceDictionary samSequenceDictionary) {
+        //TODO pass a list of intervals and add progress meter
+        //ProgressMeter progressMeter = new ProgressMeter(1.0);
+        //progressMeter.start();
+        this.outputWriter = Utils.nonNull(outputWriter);
+        this.integerCopyNumberStateCollection = Utils.nonNull(integerCopyNumberStateCollection);
+        Utils.validate(integerCopyNumberStateCollection.size() > 2, "There must be at least 3 copy number states present");
+        this.alleles = integerCopyNumberStateCollection.getAlleles();
+        this.sampleName = sampleName;
+        this.samSequenceDictionary = Utils.nonNull(samSequenceDictionary);
+    }
+
+    /**
+     * Compose the header of the variant context
+     *
+     * @param commandLine command line of the VCF generation tool
+     */
+    void composeVariantContextHeader(@Nullable final String commandLine) {
+        final VCFHeader result = new VCFHeader(Collections.emptySet(), Arrays.asList(sampleName));
+
+        /* add VCF version */
+        result.addMetaDataLine(new VCFHeaderLine(VCFHeaderVersion.VCF4_2.getFormatString(),
+                VCFHeaderVersion.VCF4_2.getVersionString()));
+
+        /* add command line */
+        if (commandLine != null) {
+            result.addMetaDataLine(new VCFHeaderLine("command", commandLine));
+        }
+
+        /* add miscellaneous header keys */
+        result.addMetaDataLine(new VCFHeaderLine(SAMPLE_NAME, sampleName));
+
+        /* header lines related to genotype formatting */
+        result.addMetaDataLine(new VCFFormatHeaderLine(CN, 1,
+                VCFHeaderLineType.Integer, "Copy number maximum a posteriori value"));
+        result.addMetaDataLine(new VCFFormatHeaderLine(CNLP, VCFHeaderLineCount.A,
+                VCFHeaderLineType.Integer, "Copy number log posterior (in Phred-scale) rounded down"));
+        result.addMetaDataLine(new VCFFormatHeaderLine(CNQ, 1,
+                VCFHeaderLineType.Integer, "Genotype call quality as the difference between the best and second best PL"));
```

Comment:
it's a little odd to use `PL` here, since we don't use it elsewhere

Reply:
Done